bdk_core/src/sparse_chain.rs

```diff
@@ -501,8 +506,8 @@ impl<I: ChainIndex> SparseChain<I> {
 /// Represents the set of changes as result of a successful [`Update`].
 #[derive(Debug, PartialEq)]
 pub struct ChangeSet<I = TxHeight> {
-    pub checkpoints: HashMap<u32, Change<BlockHash>>,
-    pub txids: HashMap<Txid, Change<I>>,
+    pub checkpoints: HashMap<u32, Option<BlockHash>>,
+    pub txids: HashMap<Txid, Option<I>>,
```

Comment:
Not a review but a conceptual question. Now that we are removing the `Change` structure, and from the test code it seems the `Option` represents the final state of the change, Is that correct? Also won't it be a useful verbosity to keep the "full change", like "From this" - "To this" pattern? Or is the thought is that can always implicitly figure out what changed from what with the final state only?

Reply:
Yes we can figure it out just by looking at the `SparseChain` and the changeset.